internal/iterbase.hpp

```diff
@@ -51,8 +51,8 @@ namespace iter {
     using AsConst = decltype(std::as_const(std::declval<T&>()));
 
     // iterator_type<C> is the type of C's iterator
-    template <typename Container>
-    using iterator_type = decltype(get_begin(std::declval<Container&>()));
+    template <typename T> //TODO: See bug https://developercommunity.visualstudio.com/content/problem/252157/sfinae-error-depends-on-name-of-template-parameter.html  for why we use T instead of Container.  Should be changed back to Container when that bug is fixed in MSVC.
```

Comment:
given that this is fixed in 16.0 I'm not sure how long this is worth keeping around